atlp-rwanda/winners-c8-bn-be
# 182210050 reset password email

**What does this PR do?**

This PR adds a function `[ passwordEmailReset() ]` in helper which helps the user who forgot the password to reset it, can be passed user details(fetch) as a parameter, then we search in an email which we will use to send the link for reset password, the link will include the user details so that we will permit him/her to change the password. This function will be used in sending a password reset link to an email after checking if the user is registered.

**Description of Task to be completed?**
Tasks to complete are:

Create an email verification helper function `(finished)`
Integrate it into the user registration route `(done)`
Document feature using swagger `Done`

**How should this be manually tested?**

Step 1. Clone this repository
Step 2. run `npm run install`
Step 3. configure the .env file in the root directory as the .env.example describes it.
The important environment variables are:

```
TECH_SUPPORT_EMAIL=
SENDGRID_API_KEY=
```
NOTE: `The TECH_SUPPORT_EMAIL` and `SENDGRID_API_KEY` variables are the email and the API Key got from `[sendgrid.com](https://sendgrid.com/)` after configuring. 

Step 4. Go to `/src/controllers/index.js/` create a controller which will fetch user details after passing the email of the user and if available  and create a link which holds the user update details searched using that entered email and  pass it to `passwordEmailReset()` as a parameter 
Step 5. run `npm run dev`
Step 6. open Postman and test that controller using the email 
Step 7. check the inbox of the email you provided in Step 4.

**Any background context you want to provide?**

This pull request is not up to date and will be finished after user registration.

**What are the relevant pivotal tracker stories?**

`#182210050`

**#### Screenshots (if appropriate)**

<img width="1057" alt="Screen Shot 2022-06-10 at 18 02 52" src="https://user-images.githubusercontent.com/82683808/173106774-80b7fa8c-53e4-4560-a3be-0500204427d7.png">


Questions:
N/A
